python/paddle/fluid/transpiler/inference_transpiler.py

```diff
@@ -54,19 +55,57 @@ def transpile(self, program, place, scope=None):
         if not isinstance(scope, core.Scope):
             raise TypeError("scope should be as Scope type or None")
         self.fuse_batch_norm(program, place, scope)
+        use_mkldnn = bool(os.getenv("FLAGS_use_mkldnn", False))
+        if use_mkldnn:
+            self.fuse_relu(program)
+
+    def fuse_relu(self, program):
+        '''
+        Transpile the program by fused relu activation for MKLDNN program.
+
+        Relu activation following batch norm OP can be fused by adding
+        'fuse_with_relu' attribute to batch norm OP.
+
+        The result of fuse is:
+            - before:
+                - batch_norm->relu->any_other_op
+            - after:
+                - batch_norm->any_other_op
+
+        :param program: program to transpile
+        :type program: Program
+        '''
+        self.block = program.block(0)
+
+        i = 0
+        while i < len(self.block.ops) - 1:
+            current_op = self.block.ops[i]
+            if current_op.type in ['batch_norm']:
+                next_op = self.block.ops[i + 1]
+                if next_op.type == 'relu':
+                    # modify bnorm OP to include relu
+                    current_op.set_attr("fuse_with_relu", True)
+                    # remove relu OP
+                    self.block.remove_op(i + 1)
+            i = i + 1
```

Comment:
- could you give some unit test to validate the accuracy of `fuse_with_relu`?
- should you call `self._remove_unused_var `, since `remove_op` will not remove variables

Reply:
done